src/state/store.ts

```diff
@@ -640,16 +640,23 @@ function promoteVariation(state: TreeState, path: number[]) {
 function setShapes(state: TreeState, shapes: DrawShape[]) {
   const node = getNodeAtPath(state.root, state.position);
   if (!node) return state;
-  const shape = shapes[0];
-  const index = node.shapes.findIndex(
-    (s) => s.orig === shape.orig && s.dest === shape.dest,
-  );
 
-  if (index !== -1) {
-    node.shapes.splice(index, 1);
+  const [shape] = shapes;
+  if (shape) {
+    const index = node.shapes.findIndex(
+      (s) => s.orig === shape.orig && s.dest === shape.dest,
+    );
+
+    if (index !== -1) {
+      node.shapes.splice(index, 1);
+    } else {
+      node.shapes.push(shape);
+    }
   } else {
-    node.shapes.push(shape);
+    node.shapes = [];
   }
+
+  return state;
```

Comment:
i assume this function is always supposed to return `state`, but this branch previously returned nothing.

Reply:
shall i change it back, or is a monomorphic return type acceptable here?